File: src/main/kotlin/br/com/fiap/techfood/application/ports/inbound/OrderInboundPort.java
package br.com.fiap.techfood.application.ports.inbound;

import br.com.fiap.techfood.application.core.domains.CartDomain;
import br.com.fiap.techfood.application.core.domains.ClientDomain;
import br.com.fiap.techfood.application.core.domains.OrderDomain;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface OrderInboundPort {

    OrderDomain save(CartDomain cartDomain, ClientDomain client);

    Optional<OrderDomain> findById(UUID id);

    List<OrderDomain> findAllApprovedOrders();

    List<OrderDomain> findAllPrepared();

    void delete(UUID id);

}

File: src/main/kotlin/br/com/fiap/techfood/application/ports/inbound/OrderInboundPort.kt
package br.com.fiap.techfood.application.ports.inbound

import br.com.fiap.techfood.application.core.domains.CartDomain
import br.com.fiap.techfood.application.core.domains.ClientDomain
import br.com.fiap.techfood.application.core.domains.OrderDomain
import java.util.*

interface OrderInboundPort {

    fun save(cartDomain: CartDomain, clientDomain: ClientDomain?): OrderDomain

    fun findById(id: UUID): Optional<OrderDomain>

    fun findAllApprovedOrders(): List<OrderDomain>

    fun findAllPrepared(): List<OrderDomain>

    fun delete(id: UUID)

    fun approvePayment(id: UUID): String;

    fun prepareOrder(id: UUID);

    fun finishOrder(id: UUID);

}
